packages/babel-plugin-proposal-destructuring-private/src/util.ts

```diff
@@ -257,12 +263,16 @@ export function* privateKeyPathIterator(pattern: t.LVal) {
   });
 }
 
+// t.LVal without t.RestElement
 type LHS =
```

Comment:
You could try using https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers